openeo-geotrellis/src/main/scala/org/openeo/geotrellis/MultiClientRangeReaderProvider.scala

```
@@ -28,8 +29,12 @@ class MultiClientRangeReaderProvider extends S3RangeReaderProvider {
     val isCloudFerro = s3Endpoint != null &&
       (s3Endpoint.toLowerCase.contains("cloudferro") || s3Endpoint.toLowerCase.endsWith(".dataspace.copernicus.eu"))
 
+    val maybeWorkspace = WorkspaceRepository.get().getWorkspaceByBucket(s3Uri.getBucket)
```

I don't believe this logic is best placed here. CreoS3Utils has helpers like `getCreoS3Client` so if we solve the problem here in MultiClientRangeReader we'd have to solve it in other places as well.  Ideally there is one factory for creating S3 clients. There are however multiple problems with `getCreoS3Client` and it's consumption:

1. It assumes the region is known or defaults to 'RegionOne'
  a. a hard coded default does not make sense imho
  b. the placeholder value would cause the endpoint to be resolved via an environment variable `SWIFT_URL` but it would not change the region. If sigv4 checking is done stricly then it would lead to authorization failures.
2. From a consumption part it often called in the same file without specifying an argument

I like the idea of a Workspace Repository. Python could provision the workspaces info that is potentially encountered.